skyvern/forge/agent.py

```diff
@@ -376,19 +378,71 @@ async def execute_step(
                 url=e.url,
                 error_message=e.error_message,
             )
-            task = await self.update_task(
+            failure_reason = f"Failed to navigate to URL. URL:{e.url}, Error:{e.error_message}"
+            try:
+                await self.fail_task(task, None, failure_reason)
+            except RetryError:
+                LOG.error(
+                    "Failed to update status and failure reason in database. Task might going to be time_out",
+                    task_id=task.task_id,
+                    step_id=step.step_id,
+                )
+            finally:
+                await self.send_task_response(
+                    task=task,
+                    last_step=step,
+                    api_key=api_key,
+                    close_browser_on_completion=close_browser_on_completion,
+                    skip_artifacts=True,
+                )
+
+            return step, detailed_output, next_step
+        except Exception as e:
+            LOG.exception(
+                "Got an unexpected exception in step, fail the task",
+                task_id=task.task_id,
+                step_id=step.step_id,
+            )
+
+            failure_reason = "unexpected exception"
+            if isinstance(e, SkyvernException):
+                failure_reason = f"unexpected SkyvernException({e.__class__.__name__})"
+
+            try:
+                await self.fail_task(task, step, failure_reason)
+            except RetryError:
+                LOG.error(
+                    "Failed to update status and failure reason in database. Task might going to be time_out",
+                    task_id=task.task_id,
+                    step_id=step.step_id,
+                )
+            finally:
+                await self.send_task_response(
```

Comment:
i think we should move this line to after line 412. it's very likely that the task is already in a timed_out or failed state. We might already have failed the task and sent a response to the webhook so we don't want to send webhook request again.

Reply:
this should be correct.